tests/check/check_h26xsigned_auth.c

```diff
@@ -882,18 +872,20 @@ START_TEST(detect_change_of_public_key)
   nalu_list_check_str(list, "GIPPGIPP");
 
   // Generate another GOP from scratch
-  nalu_list_t *second_list = create_signed_nalus("IPPPI", settings[_i]);
-  nalu_list_check_str(second_list, "GIPPPGI");
+  // This will get a new public key
+  nalu_list_t *list_with_new_public_key = create_signed_nalus("IPPPI", settings[_i]);
+  nalu_list_check_str(list_with_new_public_key, "GIPPPGI");
 
-  nalu_list_append_and_free(list, second_list);
+  nalu_list_append_and_free(list, list_with_new_public_key);
   nalu_list_check_str(list, "GIPPGIPPGIPPPGI");
 
-  // One pending NALU per GOP. Note that the mid GOP (IPPGI) includes the reset on the camera. It
-  // will be marked as invalid and compute 3 more NALUs than expected. In G it is communicated
-  // there is only 2 NALUs present (GI). So missed NALUs equals -3 (IPP).
-  const struct validation_stats expected = {
-      .valid_gops = 2, .invalid_gops = 2, .missed_nalus = -3, .pending_nalus = 4, .public_key_has_changed = true
-  };
+  // One pending NALU per GOP. It will be marked as invalid and compute 3 more NALUs than expected.
+  // In G it is communicated there is only 2 NALUs present (GI). So missed NALUs equals -3 (IPP).
```

Comment:
May I suggest something like this
```
// The list will be validated successfully up to the third SEI (G) which has the new Public key.
//
//   GI      -> .P (valid, 1 pending, public_key_has_changed = false)
//   IPPGI   -> ....P (valid, 1 pending, public_key_has_changed = false)
//   IPPG*I  -> NNN.P (invalid, 1 pending, public_key_has_changed = true, -3 missing)
//   IPPPG*I -> N....P (invalid, 1 pending, public_key_has_changed = false)
// where G* has the new Public key. Note that we get -3 missing since we receive 3 more than what is expected according to G*.
```